tools/iscc-sum/iscc_similarity.xml

```diff
@@ -0,0 +1,211 @@
+<tool id="iscc_sum_compare" name="Compare ISCC hash similarity" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="24.1">
+    <description>with ISCC-SUM</description>
+    <macros>
+        <import>macros.xml</import>
+        <import>creators.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <expand macro="version_command" />
+    <creator>
+        <expand macro="creators/iscc" />
+        <expand macro="creators/lco" />
+        <expand macro="creators/maartenpaul" />
+        <expand macro="creators/etzm" />
+    </creator>
+        
+    <command detect_errors="exit_code"><![CDATA[
+        #if $input_type.input_selector == "two_files":
+            ## Pairwise file comparison
+            iscc-sum --similar --threshold $threshold '$input_type.file1' '$input_type.file2' > '${output_file}'
+        #elif $input_type.input_selector == "two_collections":
+            ## Two collections comparison - generate combined ISCC for each
+            ISCC1=\$(cat
+            #for $file in $input_type.collection1:
+                '$file'
+            #end for
+            | iscc-sum - | cut -d':' -f2 | cut -d' ' -f1) &&
+
+            ISCC2=\$(cat
+            #for $file in $input_type.collection2:
+                '$file'
+            #end for
+            | iscc-sum - | cut -d':' -f2 | cut -d' ' -f1) &&
+
+            ## Create temp files for comparison
+            echo \$ISCC1 > temp1.txt &&
+            echo \$ISCC2 > temp2.txt &&
+
+            ## Use iscc-sum to compare similarity and output directly
+            iscc-sum --similar --threshold $threshold temp1.txt temp2.txt > '${output_file}' || true
+        #else:
+            ## Single collection - find similar files within
+            mkdir -p input_files &&
+            #for $file in $input_type.file_collection:
+                ln -s '$file' 'input_files/$file.element_identifier' &&
+            #end for
+            iscc-sum --similar --threshold $threshold input_files/* > '${output_file}'
+        #end if
+    ]]></command>
+
+    <inputs>
+        <conditional name="input_type">
+            <param name="input_selector" type="select" label="Input type">
+                <option value="two_files">Compare two files</option>
+                <option value="collection">Find similar files in collection</option>
```

Comment:
I see that internally this is about files, but from a Galaxy user's perspective, there are only datasets and collections thereof:

```suggestion
                <option value="two_files">Compare two datasets</option>
                <option value="collection">Find similar datasets in collection</option>
```